bundles/tools.mdsd.ecoreworkflow.mwe2lib/src/tools/mdsd/ecoreworkflow/mwe2lib/component/RegexComponent.java

```diff
@@ -32,75 +34,165 @@
+            for (Mapping replacement : replacements) {
+                content = content.replaceAll(replacement.getFrom(), replacement.getTo());
+            }
+            writeFile(uri, content);
+            LOG.info("Regex Replacement for:" + uri.toString());
+        }
+    }
+
+    protected String readFile(URI fileUri) throws IOException {
+        try (var stream = uriConverter.createInputStream(fileUri)) {
+            return new String(stream.readAllBytes(), charset);
+        }
+    }
+
+    protected void writeFile(URI fileUri, String content) throws IOException {
+        try (var stream = uriConverter.createOutputStream(fileUri)) {
+            stream.write(content.getBytes(charset));
+        }
+    }
+
+    protected List<URI> determineFilesToReplace(Replacement replacement) throws IOException {
+        Optional<URI> directory = Optional.ofNullable(replacement.getDirectory())
+            .map(this::fromFilename)
+            .map(this::ensureAbsoluteURI)
+            .map(this::ensureEmptyLastSegment);
+
+        List<URI> filesToProcess = replacement.getFilenames()
+            .stream()
+            .map(this::fromFilename)
+            .map(uri -> directory.map(uri::resolve)
+                .orElse(uri))
+            .collect(Collectors.toList());
+
+        if (replacement.getWildcard() != null && directory.isPresent()) {
+            final PathMatcher matcher = FileSystems.getDefault()
+                .getPathMatcher("glob:" + replacement.getWildcard());
+
+            var dirPath = getAbsolutePathOfDirectory(directory.get());
+
+            Files.walkFileTree(dirPath, new SimpleFileVisitor<Path>() {
+                @Override
+                public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                    var relativeMatch = dirPath.relativize(file);
+                    if (matcher.matches(relativeMatch)) {
+                        filesToProcess.add(URI.createURI(file.toUri()
+                            .toString()));
+                    }
+                    return FileVisitResult.CONTINUE;
+                }
+
+                @Override
+                public FileVisitResult visitFileFailed(Path file, IOException exc) throws IOException {
+                    return FileVisitResult.CONTINUE;
+                }
+            });
+        }
+        return filesToProcess;
+    }
+
+    protected URI fromFilename(String filename) {
+        var uri = parseURI(filename);
+        return uri.orElseGet(() -> URI.createURI(Path.of(filename)
+            .toUri()
+            .toString()));
+    }
+
+    protected Optional<URI> parseURI(String uri) {
+        try {
+            var jUri = new java.net.URI(uri);
+            return Optional.of(URI.createURI(jUri.toString()));
+        } catch (URISyntaxException e) {
+            return Optional.empty();
+        }
+
+    }
+
+    protected Path getAbsolutePathOfDirectory(URI uri) {
+        Path result;

```

Comment:
Conditions of control structures might become simpler if `result =` is replaced by `return`.

Reply:
Ok. Saves one line. Conditions did not change.